app/services/list_collection.rb

```diff
@@ -1,9 +1,10 @@
 class ListCollection
+  include ActiveModel::Model
 
-  def initialize page: 1, page_size: 25
```

Comment:
Interesting choice switching to `ActiveModel::Model` to handle the different ways you want this initialized.
That's the first time I've seen someone tackle the problem that way.